columnflow/production/normalization.py

```diff
@@ -76,31 +173,41 @@ def normalization_weights_setup(
     Sets up objects required by the computation of normalization weights and stores them as instance
     attributes:
 
-        - py:attr:`selection_stats`: The stats dict loaded from the output of MergeSelectionsStats.
         - py:attr:`sum_weights_table`: A sparse array serving as a lookup table for the sum of event
           weights per process id.
         - py:attr:`xs_table`: A sparse array serving as a lookup table for cross sections of all
           processes known to the config of the task, with keys being process ids.
     """
     # load the selection stats
-    selection_stats = self.task.cached_value(
-        key="selection_stats",
-        func=lambda: inputs["selection_stats"]["collection"][0]["stats"].load(formatter="json"),
-    )
+    normalization_selection_stats = {
+        dataset: self.task.cached_value(
+            key=f"selection_stats_{dataset.name}",
+            func=lambda: inp["collection"][0]["stats"].load(formatter="json"),
+        )
+        for dataset, inp in inputs[self.selection_stats_key].items()
```

Comment:
Hey @riga, just to make sure: this will cache the selection stats per dataset, therefore even though we run this setup function multiple times (both for stitched and unstitched normalization), the cached values will still not produce any unexpected behaviour, right?

Reply:
I checked and after a small bugfix, it ran through and the results were the same as before :+1: